storage/acceptance/files_test.rb

```diff
@@ -403,6 +403,29 @@ def mock_cipher.random_key
     end
   end
 
+  describe "generate_signed_post_policy_v4" do
+    it "generates a v4 signed post policy v4 for a file in a bucket" do
+      refute bucket.file remote_file_name
+      timestamp = Time.new 2020, 1, 1, 0, 0, 0, "+00:00"
+      out = nil
+      Time.stub :now, timestamp do
+        out, _err = capture_io do
+          generate_signed_post_policy_v4 bucket_name: bucket.name,
+                                         file_name:   remote_file_name
+        end
+      end
+      assert_includes out, "<form action='https://storage.googleapis.com/#{bucket.name}/' method='POST' enctype='multipart/form-data'>"
+      assert_includes out, "<input name='x-goog-meta-test' value='data' type='hidden'/>"
+      assert_includes out, "<input name='key' value='#{remote_file_name}' type='hidden'/>"
+      assert_includes out, "<input name='x-goog-date' value='20200101T000000Z' type='hidden'/>"
+      assert_includes out, "<input name='x-goog-credential' value='664476911263-rfrsgaitrs4niii0k60ns485cu4vgm2u@developer.gserviceaccount.com/20200101/auto/storage/goog4_request' type='hidden'/>"
```

Comment:
@frankyn This line will fail unless the test is run with my development credentials. I'm not sure of the best way around this, but I was thinking that an end-to-end integration test using an actual HTTP POST would be good, like [the test I added in google-cloud-ruby](https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-storage/acceptance/storage/bucket_generate_signed_post_policy_v4_test.rb#L32-L52). However, I'm not sure how to return the `post_object` from `files.rb`. All existing methods only return string output via `puts` and `out`. Any suggestions?

Reply:
We are only confirming the existing of element in form and not the value correctness. 

You don't need to stub time in this case.

Here's Node.js test:
https://github.com/googleapis/nodejs-storage/blob/master/samples/system-test/files.test.js#L243-L263

Reply:
OK, I'll update to match the Node test. Thanks!